examples/src/main/java/org/bitcoinj/examples/RestoreFromSeedThenDump.java

```diff
@@ -0,0 +1,117 @@
+import org.bitcoinj.wallet.DeterministicKeyChain;
+import org.bitcoinj.wallet.DeterministicSeed;
+import org.bitcoinj.wallet.KeyChainGroup;
+import org.bitcoinj.wallet.Wallet;
+import org.bitcoinj.wallet.authentication.AuthenticationGroupExtension;
+
+import java.io.File;
+
+/**
+ * The following example shows you how to restore a HD wallet from a previously generated deterministic seed.
+ * In this example we manually setup the blockchain, peer group, etc. You can also use the WalletAppKit which provides a restoreWalletFromSeed function to load a wallet from a deterministic seed.
+ */
+public class RestoreFromSeedThenDump {
+
+    public static void main(String[] args) throws Exception {
+        BriefLogFormatter.init();
+        NetworkParameters params = TestNet3Params.get();
+        if (args.length == 0) {
+            System.out.println("Enter a seed phrase");
+        }
+
+        // Bitcoinj supports hierarchical deterministic wallets (or "HD Wallets"): https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+        // HD wallets allow you to restore your wallet simply from a root seed. This seed can be represented using a short mnemonic sentence as described in BIP 39: https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki
+
+        // Here we restore our wallet from a seed with no passphrase. Also have a look at the BackupToMnemonicSeed.java example that shows how to backup a wallet by creating a mnemonic sentence.
+        String seedCode = args[0];
+        String passphrase = "";
+        long creationtime = 1547463771L;//1409478661L;
+
+        DeterministicSeed seed = new DeterministicSeed(seedCode, null, passphrase, creationtime);
+        DerivationPathFactory factory = DerivationPathFactory.get(params);
+        KeyChainGroup keyChainGroup = KeyChainGroup.builder(params)
+                .addChain(DeterministicKeyChain.builder()
+                        .seed(seed)
+                        .accountPath(factory.bip32DerivationPath(0))
+                        .build())
+                .addChain(DeterministicKeyChain.builder()
+                        .seed(seed)
+                        .accountPath(factory.bip44DerivationPath(0))
+                        .build())
+                .build();
+
+        // The wallet class provides a easy fromSeed() function that loads a new wallet from a given seed.
+        Wallet wallet = new Wallet(params, keyChainGroup);
+        AuthenticationGroupExtension mnext = new AuthenticationGroupExtension(wallet);
+        mnext.addKeyChain(seed, factory.masternodeOwnerDerivationPath(), AuthenticationKeyChain.KeyChainType.MASTERNODE_OWNER);
+        mnext.addKeyChain(seed, factory.masternodeVotingDerivationPath(), AuthenticationKeyChain.KeyChainType.MASTERNODE_VOTING);
+        mnext.addKeyChain(seed, factory.masternodeOperatorDerivationPath(), AuthenticationKeyChain.KeyChainType.MASTERNODE_OPERATOR);
+        wallet.addExtension(mnext);
```

Comment:
This is one of the ways that we add masternode key support to the wallet.  For now, this support is not included by default.

When we get to the point of adding the Masternode Key support to Dash Wallet, we may rethink this.  As with CoinJoin and Platform features with DashPay, the wallet has to be created/loaded with features activated or initialized.